designer/server/src/main/scala/pl/touk/nussknacker/ui/api/ManagementApiEndpoints.scala

```diff
@@ -0,0 +1,60 @@
+package pl.touk.nussknacker.ui.api
+
+import pl.touk.nussknacker.engine.api.CirceUtil._
+import derevo.circe.{decoder, encoder}
+import derevo.derive
+import pl.touk.nussknacker.ui.validation.CustomActionValidationError
+import pl.touk.nussknacker.engine.api.process.ProcessName
+import pl.touk.nussknacker.restmodel.{BaseEndpointDefinitions, CustomActionRequest}
+import pl.touk.nussknacker.restmodel.BaseEndpointDefinitions.SecuredEndpoint
+import pl.touk.nussknacker.security.AuthCredentials
+import sttp.tapir.{EndpointInput, derevo, path, statusCode, _}
+import TapirCodecs.ScenarioNameCodec._
+import sttp.model.StatusCode.Ok
+import sttp.tapir.generic.auto._
+import sttp.tapir.json.circe._
+import io.circe.generic.auto._
+import pl.touk.nussknacker.engine.api.context.PartSubGraphCompilationError
+import pl.touk.nussknacker.engine.deployment.CustomActionValidationResult
+import pl.touk.nussknacker.restmodel.validation.ValidationResults.NodeValidationError
+import sttp.model.StatusCode
+import sttp.tapir.generic.auto._
+import sttp.tapir.json.circe.jsonBody
+import sttp.tapir.derevo.schema
+
+class ManagementApiEndpoints(auth: EndpointInput[AuthCredentials]) extends BaseEndpointDefinitions {
+
+  private lazy val baseProcessManagementEndpoint = baseNuApiEndpoint.in("processManagement")
+
+  lazy val customActionValidationEndpoint: SecuredEndpoint[
+    (ProcessName, CustomActionRequest),
+    CustomActionValidationError,
```

Comment:
See comments in https://github.com/TouK/nussknacker/commit/7cd849ac0b77e2b0cdae606a84c2757b4fca66e9